pipreqs/pipreqs.py

```diff
@@ -137,70 +158,93 @@ def get_all_imports(
         # Cleanup: We only want to first part of the import.
         # Ex: from django.conf --> django.conf. But we only want django
         # as an import.
-        cleaned_name, _, _ = name.partition('.')
+        cleaned_name, _, _ = name.partition(".")
         imports.add(cleaned_name)
 
     packages = imports - (set(candidates) & imports)
-    logging.debug('Found packages: {0}'.format(packages))
+    logging.debug("Found packages: {0}".format(packages))
 
     with open(join("stdlib"), "r") as f:
         data = {x.strip() for x in f}
 
     return list(packages - data)
 
 
+def filter_line(line):
+    return len(line) > 0 and line[0] != "#"
+
+
+def filter_ext(file_name, acceptable):
+    return os.path.splitext(file_name)[1] in acceptable
+
+
+def ipynb_2_py(file_name, encoding=None):
+    """
+
+    Args:
+        file_name (str): notebook file path to parse as python script
+        encoding  (str): encoding of file
+
+    Returns:
+        str: parsed string
+
+    """
+
+    exporter = PythonExporter()
+    (body, _) = exporter.from_filename(file_name)
+
+    return body.encode(encoding if encoding is not None else "utf-8")
```

Comment:
done

Reply:
Setting the default value to utf-8 makes the tests fail. It happens because, when calling the `init` function without an encoding option, it defaults to _None_, passing _None_ as an argument to to the `get_all_imports` function, which passes _None_ as the encoding argument to the `ipynb_2_py` function. Leaving the function working in this way, makes it capable of dealing with _None_ values as input without crashing the program.

Reply:
@Fernando-crz then why are we passing None at all? why not always default to the python standar that is utf-8? I think a new commit to change this pipreqs wide could be good idea

Reply:
Done